src/mm/mod.rs

```diff
@@ -107,3 +103,19 @@ impl<'alloc, T: arch::ArchitectureMemory> MemoryManager<'alloc, T> {
         self.arch.reload();
     }
 }
+
+pub trait MemoryManager {
+    fn map(&mut self, to: usize, from: usize, perms: Permissions);
+    fn reload_pt(&mut self);
+
+}
```

Comment:
Could you put this on top of the file? I prefer having first the trait, then the impl

Reply:
done